googleapiclient/discovery.py

```diff
@@ -171,6 +173,9 @@ def build(serviceName,
       request.
     credentials: oauth2client.Credentials, credentials to be used for
       authentication.
+    cache_discovery: Boolean, whether or not cache the discovery doc.
+    cache: Object for caching discovery documents. See modules in
```

Comment:
How well do you know the abc standard library module? Defining an interface class for the multiple cache implementations would let you properly state the type of this parameter.